server/src/test/scala/drt/shared/FlightDisplayFilterSpec.scala

```diff
@@ -0,0 +1,129 @@
+  val redListT4postT4 = ApiFlightWithSplits(ArrivalGenerator.arrival(sch = afterT4Opening, origin = redListOriginInBolivia, terminal = T4), Set())
+  val nonRedListT4postT4 = ApiFlightWithSplits(ArrivalGenerator.arrival(sch = afterT4Opening, origin = nonRedListOriginInFrance, terminal = T4), Set())
+  val redListT5postT4 = ApiFlightWithSplits(ArrivalGenerator.arrival(sch = afterT4Opening, origin = redListOriginInBolivia, terminal = T5), Set())
+  val nonRedListT5postT4 = ApiFlightWithSplits(ArrivalGenerator.arrival(sch = afterT4Opening, origin = nonRedListOriginInFrance, terminal = T5), Set())
+
+
+  val filter = LhrFlightDisplayFilter(isRedListOrigin, t4OpeningMillis)
+
+  "Given flights for LHR spanning all terminals arriving before T4 starts handling red list flights" >> {
+    val flightsPreT4Opening = List(redListT2preT4, nonRedListT2preT4, redListT3preT4, nonRedListT3preT4, redListT4preT4, nonRedListT4preT4, redListT5preT4, nonRedListT5preT4)
+    "When I ask for T2 display flights, I should get only T2 flights" >> {
+      filter.forTerminal(flightsPreT4Opening, T2) === List(redListT2preT4, nonRedListT2preT4)
+    }
+    "When I ask for T3 display flights, I should get all the T3 flights, plus any T2 & T5 red list origin flights" >> {
+      filter.forTerminal(flightsPreT4Opening, T3) === List(redListT2preT4, redListT3preT4, nonRedListT3preT4, redListT5preT4)
+    }
+    "When I ask for T4 display flights, I should get only T4 flights" >> {
+      filter.forTerminal(flightsPreT4Opening, T4) === List(redListT4preT4, nonRedListT4preT4)
+    }
+    "When I ask for T5 display flights, I should get only T5 flights" >> {
+      filter.forTerminal(flightsPreT4Opening, T5) === List(redListT5preT4, nonRedListT5preT4)
+    }
+  }
+
+  "Given flights for LHR spanning all terminals arriving after T4 starts handling red list flights" >> {
+    val flightsPostT4Opening = List(redListT2postT4, nonRedListT2postT4, redListT3postT4, nonRedListT3postT4, redListT4postT4, nonRedListT4postT4, redListT5postT4, nonRedListT5postT4)
+    "When I ask for T2 display flights, I should get only T2 flights" >> {
+      filter.forTerminal(flightsPostT4Opening, T2) === List(redListT2postT4, nonRedListT2postT4)
+    }
+    "When I ask for T3 display flights, I should get only T3 flights" >> {
+      filter.forTerminal(flightsPostT4Opening, T3) === List(redListT3postT4, nonRedListT3postT4)
+    }
+    "When I ask for T4 display flights, I should get all the T4 flights, plus any T2 & T5 red list origin flights" >> {
+      filter.forTerminal(flightsPostT4Opening, T4) === List(redListT2postT4, redListT4postT4, nonRedListT4postT4, redListT5postT4)
+    }
+    "When I ask for T5 display flights, I should get only T5 flights" >> {
+      filter.forTerminal(flightsPostT4Opening, T5) === List(redListT5postT4, nonRedListT5postT4)
+    }
+  }
+}
+
+object ArrivalGenerator {
```

Comment:
I would prefer to ArrivalGenerator to move to other file as it not just related to this test .